### DISCUSSION TITLE
How to Check the Actual Visibility of a Child Widget

### DISCUSSION THREAD

**Discussion by recsater-uf** [Textualize/textual#5287] [Q&A] [answered]
When Container A yields Widget B and A has display: none, B.display still remains True.

However, B is not actually rendered in this case. How can I accurately determine whether B is truly visible or not?

Should we update `@property def display(self)`?

**Comment by willmcgugan** [comment, marked as answer]
Widgets have an `is_on_screen` property, which will be `True` if they were shown in the last frame.

**Reply by recsater-uf** [reply]
You are a genius. Thank you so much.
